src/GT-Playground/GTPlayPageFilePersistence.class.st

```diff
@@ -6,7 +6,12 @@ Class {
 
 { #category : #private }
 GTPlayPageFilePersistence >> deleteFromFileSystem [
-	self fileReference ensureDelete
+
+	| aFileReference |
+	
+	aFileReference := self fileReference.
+	(aFileReference notNil and: [ aFileReference fileSystem isWritable ]) ifTrue: [ 
+		aFileReference ensureDelete ]
```

Comment:
Also an idea (maybe not for this work). 
It would be really nice to avoid such checks in all places where fileReferences are using.
Specially configured file reference could be a deaf object and ignore all write accesses. So users like this one will only need to check location of cache in one place and replace it with something like
```Smalltalk
cacheDirectory := cacheDirectory asIgnoredAnyWrites.
```

Reply:
Maybe but I would solve it in a different PR because this may have strange consequences when data may be lost without any notice